src/general-components/API/SettingsCache.ts

```diff
@@ -0,0 +1,117 @@
+import {PaginationLoader} from "./PaginationLoader";
+import {SettingResource, UserSettingResource} from "../Datastructures";
+import * as Settings from "../../general-components/API/calls/Settings";
+import {SettingsList, UserSetting} from "../Settings/SettingsList";
+
+
+/**
+ * Speichert alle Einstellungsmöglichkeiten und aktualisiert diese regelmäßig
+ */
+export class SettingsCache {
+
+
+    private _userId: number;
+    /**
+     * last time settings did get updated
+     * @private
+     */
+    private _lastLoad: Date;
+    /**
+     * all settings with the current value
+     * @private
+     */
+    private _userSettings: SettingsList;
+
+    private settingsLoader: PaginationLoader<SettingResource>;
+    private userSettingsLoader: PaginationLoader<UserSettingResource>;
+
+    constructor(token: string, userId: number) {
+        this._userId = userId;
+        this._lastLoad = new Date(0);
+        this._userSettings = new SettingsList();
+
+        this.settingsLoader = new PaginationLoader<SettingResource>((page) => {
+            return Settings.getSettings(token, page)
+        })
+        this.userSettingsLoader = new PaginationLoader<UserSettingResource>((page) => {
+            return Settings.getUserSettings(userId, token, page)
+        })
+
+    }
+
+    /**
+     * Lädt alle Daten aus dem Backend, wenn nicht anders angegeben, werden keine Caches berücksichtigt
+     * @param cachedSettings Ob die Einstellungen auch aus dem Cache geladen werden dürfen
+     * @param cachedUserSettings Ob die User Einstellungen auch aus dem Cache geladen werden dürfen
+     * @private
+     */
+    private async loadData(cachedSettings: boolean = false, cachedUserSettings: boolean = false) {
+        // Promise.all damit die Backend calls gleichzeitig stattfinden können und nicht nacheinander passieren müssen
+        const [settings, userSettings] = await Promise.all([this.settingsLoader.getAll(cachedSettings), this.userSettingsLoader.getAll(cachedUserSettings)]);
+        return settings.map((s) => {
+            let userS = userSettings.find((su) => su.setting_id === s.id);
+            let v = s.default;
+            const exists = !!userS && !!userS.value;
+            if (exists) {
+                //TODO make beautiful
+                v = userS?.value ?? v;
```

Comment:
but indeed, this should somehow be made better. maybe move the exists condition into the if or find a workaround.

Reply:
fixed the logic, now the variable wouldn't make sense anyway